app/controllers/admin/api/objects_controller.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class Admin::Api::ObjectsController < Admin::Api::BaseController
+
+  class AllowedStatusObject
```

Comment:
This is an "authorizer", right? How about, for example:

```suggestion
  class Authorizer
```

...and of course also name the children classes `<ObjectType>Authorizer`.

It could be more literal too of course, like `StatusQueryAuthorizer`, if you prefer.

Reply:
true true.